src/gaussnewton.jl

```diff
@@ -81,15 +81,31 @@ function SciMLBase.__init(prob::NonlinearLeastSquaresProblem{uType, iip}, alg_::
     kwargs...) where {uType, iip}
     alg = get_concrete_algorithm(alg_, prob)
     @unpack f, u0, p = prob
+
+    # Use QR if the user did not specify a linear solver
+    if alg.linsolve === nothing || alg.linsolve isa QRFactorization ||
+       alg.linsolve isa FastQRFactorization
```

Comment:
This doesn't make sense. It should be open so other methods like GMRES apply.

Reply:
GMRES works on non-square? https://github.com/JuliaSmoothOptimizers/Krylov.jl/blob/main/src/gmres.jl

Reply:
is there a `can_handle_non_square` trait?